clients/src/main/java/org/apache/kafka/clients/consumer/internals/AsyncKafkaConsumer.java

```diff
@@ -1279,7 +1281,7 @@ private void releaseAssignmentAndLeaveGroup(final Timer timer) {
         UnsubscribeEvent unsubscribeEvent = new UnsubscribeEvent(calculateDeadlineMs(timer));
         applicationEventHandler.add(unsubscribeEvent);
         try {
-            processBackgroundEvents(unsubscribeEvent.future(), timer);
+            processBackgroundEvents(unsubscribeEvent.future(), timer, e -> e instanceof InvalidTopicException);
```

Comment:
Is there a chance that the `InvalidTopicException` could be nested inside another exception?

Reply:
No I would say. It's already a KafkaException (we do wrap what it's not), and it's simply created here and passed into the event on the other link you shared.
https://github.com/apache/kafka/blob/18340c973378610cd5930974201088b6aca8a8ad/clients/src/main/java/org/apache/kafka/clients/Metadata.java#L648